### PR TITLE
update source-s3 spec to make keys optional

### PR DESCRIPTION
**Pull request by psFried**
Resolves points 2 and 3 in #181

### CODE REVIEW

File: source-s3/main.go [hunk 1]
```diff
@@ -255,7 +255,7 @@ func main() {
 			"region": {
 				"type":        "string",
 				"title":       "AWS Region",
-				"description": "The name of the AWS region where the S3 stream is located",
+				"description": "The name of the AWS region where the S3 stream is located. \"us-east-1\" is a popular default you can try, if you're unsure what to put here.",
```
Comment (saterus) [inline source-s3/main.go:258]: Is "S3 stream" the way _we're_ choosing to talk about it here? Or would this make more sense to the user if we just said "S3 bucket"?

File: source-s3/main.go [hunk 2]
```diff
@@ -209,8 +209,8 @@ func main() {
 		"title":   "S3 Source Spec",
 		"type":    "object",
 		"required": [
-			"awsAccessKeyId",
-			"awsSecretAccessKey"
```
Comment (saterus) [inline source-s3/main.go:213]: Should we add anything else to the `description` fields for these? Omitting these fields is only really viable if it's completely public, right?

**Review by saterus — APPROVED** [review]
Looks good. Small questions, but not blockers.

File: source-s3/main.go (hunk 1 above)
Comment (psFried) [reply source-s3/main.go]: lol I didn't even notice that. I'm assuming it's an artifact of copypasta from source-kinesis. Changed it to "bucket".

File: source-s3/main.go (hunk 2 above)
Comment (psFried) [reply source-s3/main.go:213]: Yeah good idea. I added a bit of clarification.

**Pull request merged**